specs/eventing/control-plane.md

```diff
@@ -0,0 +1,985 @@
+Triggers. When the Broker's `Ready` condition is `true`, the Broker MUST provide
+a `status.address.url` which accepts all valid CloudEvents and MUST attempt to
+forward the received events for filtering to each associated Trigger whose
+`Ready` condition is `true`. As described in the
+[Trigger Lifecycle](#trigger-lifecycle) section, a Broker MAY forward events to
+an associated Trigger which which does not currently have a `true` `Ready`
+condition, including events received by the Broker before the Trigger was
+created.
+
+The annotation `eventing.knative.dev/broker.class` MAY be used to select a
+particular implementation of a Broker. When a Broker is created, the
+`eventing.knative.dev/broker.class` annotation and the `spec.config` field MUST
+be populated (`spec.config` MAY be an empty object) to indicate which of several
+possible Broker implementations to use. It is RECOMMENDED to default the
+`eventing.knative.dev/broker.class` field on creation if it is unpopulated. Once
+created, both fields MUST be immutable; the Broker MUST be deleted and
+re-created to change the implementation class or `spec.config`. This pattern is
+chosen to make it clear that changing the implementation class or `spec.config`
+is not an atomic operation and that any implementation would be likely to result
+in event loss during the transition.
+
+### Trigger Lifecycle
+
+The lifecycle of a Trigger is independent of the Broker it refers to in its
+`spec.broker` field; if the Broker does not currently exist or the Broker's
+`Ready` condition is not `true`, then the Trigger's `Ready` condition MUST be
+`false`, and the reason SHOULD indicate that the corresponding Broker is missing
+or not ready.
+
+The Trigger's controller MUST also set the `status.subscriberUri` field based on
+resolving the `spec.subscriber` field before setting the `Ready` condition to
+`true`. If the `spec.subscriber.ref` field points to a resource which does not
+exist or cannot be resolved via
+[Addressable resolution](#addressable-resolution), the Trigger MUST set the
+`Ready` condition to `false`, and at least one condition should indicate the
+reason for the error.
+
+If the Trigger's `spec.delivery.deadLetterSink` field it set, it MUST be
+resolved to a URI and reported in `status.deadLetterSinkUri` in the same manner
+as the `spec.subscriber` field before setting the `Ready` condition to `true`.
+
+Once created, the Trigger's `spec.broker` SHOULD NOT permit updates; to change
+the `spec.broker`, the Trigger can instead be deleted and re-created. This
+pattern is chosen to make it clear that changing the `spec.broker` is not an
+atomic operation, as it could span multiple storage systems. Changes to
+`spec.subscriber`, `spec.filter` and other fields SHOULD be permitted, as these
+could occur within a single storage system.
+
+When a Trigger becomes associated with a Broker (either due to creating the
+Trigger or the Broker), the Trigger MUST only set the `Ready` condition to
```

Comment:
Above we talk about how we expect the Broker and Trigger to be managed by the same controller, but what if they're not? Are we really mandating two controllers sync themselves with each other? Why isn't the status of the Trigger solely based on whether it can connect to a Broker and it's Subscriber?

Reply:
Or put it another way... what bad thing would happen if we removed this MUST?